commit b0a6df6fbd9117b41a7f4e3bc861e20fbadb1956
Author: Olivier Goffart <ogoffart@woboq.com>
Date:   Tue Feb 17 15:24:08 2015 +0100

    Fix QIcon::fromTheme(xxx, someFallback) would not return the fallback
    
    because KIconEngine::availableSize always return something even if the icon
    is not there
    
    BUG: 342906
    REVIEW: 122608

diff --git a/src/kiconengine.cpp b/src/kiconengine.cpp
index 530403e..6dff533 100644
--- a/src/kiconengine.cpp
+++ b/src/kiconengine.cpp
@@ -111,6 +111,15 @@ QList<QSize> KIconEngine::availableSizes(QIcon::Mode mode, QIcon::State state) c
 {
     Q_UNUSED(mode);
     Q_UNUSED(state);
+
+    if (!mIconLoader) {
+        return QList<QSize>();
+    }
+
+    if (mIconLoader->iconPath(mIconName, KIconLoader::Desktop, KIconLoader::MatchBest).isEmpty()) {
+        return QList<QSize>();
+    }
+
     return QList<QSize>() << QSize(16, 16)
            << QSize(22, 22)
            << QSize(32, 32)
